include/knowhere/index/index_node.h

```diff
@@ -290,13 +294,17 @@ class IndexNode : public Object {
         futs.reserve(nq);
         if (retain_iterator_order) {
             for (size_t i = 0; i < nq; i++) {
-                futs.emplace_back(
-                    ThreadPool::GetGlobalSearchThreadPool()->push([&, idx = i]() { task_with_ordered_iterator(idx); }));
+                futs.emplace_back(ThreadPool::GetGlobalSearchThreadPool()->push([&, idx = i]() {
+                    ThreadPool::ScopedSearchOmpSetter setter(1);
```

Comment:
Can we rewrite the push to include this omp setter instead of calling it everywhere

Reply:
@cqy123456 Is this feasible? Are there any scenarios where we do not need the `OmpSetter`?